synapseclient/models/mixins/table_components.py

```diff
@@ -1770,7 +1770,55 @@ def _construct_partial_rows_for_upsert(
             column_id = entity.columns[column].id
             column_type = entity.columns[column].column_type
             cell_value = matching_row[column].values[0]
-            if not hasattr(row, column) or cell_value != getattr(row, column):
+
```

Comment:
Do you know why this part is needed even after using `.convert_dtypes()`? and Dan mentioning that the Synapse Tables uploaded results were OK after adding `convert_dtypes()` in her investigation. Is this due to `itertuples` coercing into native pandas dtypes and so we need the explicit handling of NAs below